contrib/build_purger.rb

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env ruby
+require 'json'
+require 'time'
+require 'optparse'
+require 'yaml'
+
+options = {}
+OptionParser.new do |opts|
+  opts.banner = "Usage: build_purger.rb [options]"
+  opts.on('--days DAYS', 'Stale point') { |v| options[:stale_point] = Time.now - v.to_i*24*60*60 }
+  opts.on('--path PATH', 'Build path') { |v| options[:path] = v }
+  opts.on('--dry-run', 'Dry run') { |v| options[:dry_run] = v }
+end.parse!
+
+abort("#{options[:path]} is not a directory") unless File.directory?(options[:path])
+
+jobs = []
+projects = Dir["#{options[:path]}/projects/*"].map { |p| p.gsub("#{options[:path]}/projects/","") }
+stale_jobs = []
+
+projects.each do |p|
+  data_path = File.join(options[:path], "data", p)
+  groups_path = File.join(data_path, "groups")
+  latest_path = File.join(data_path, "latest")
+  # jobs that have a symlink either as a group or as latest won't be deleted
+  groups_and_latest = []
+  group_jobs = Dir["#{groups_path}/*"]
+  groups_and_latest << `readlink #{Dir["latest_path"]}`
```

Comment:
You can use `File.readlink()` and avoid the shell escape.

Reply:
good catch, fixed